test/index.js

```diff
@@ -56,76 +56,78 @@ describe('normalize', function() {
     done();
   });
 
-  it('calls the type function to check validity', function(done) {
-    var called = false;
-    var type = function() {
-      called = true;
-      return false;
-    };
-    var value = 1;
-    var result = normalize(type, value);
-    expect(result).toEqual(null);
-    expect(called).toEqual(true);
+  it('calls the type function to attempt coercion', function(done) {
+    var expected = 1;
+    var type = expect.createSpy().andCall(function(value) {
+      return value;
+    });
+    var result = normalize(type, expected);
+    expect(result).toEqual(expected);
+    expect(type).toHaveBeenCalled();
     done();
   });
 
   it('calls the value if it is a function', function(done) {
     var type = 'string';
     var expected = 'test string';
-    var value = function() {
+    var value = expect.createSpy().andCall(function() {
       return expected;
-    };
+    });
     var result = normalize(type, value);
     expect(result).toEqual(expected);
+    expect(value).toHaveBeenCalled();
     done();
   });
 
-  it('checks the result of function against predicate', function(done) {
+  it('checks the result of function against coercer', function(done) {
     var expected = 'test string';
-    var called = false;
-    var predicate = function(value) {
-      called = true;
-      return (typeof value === 'string');
-    };
-    var value = function() {
+    var coercer = expect.createSpy().andCall(function(value) {
+      return (typeof value === 'string') ? value : null;
+    });
+    var value = expect.createSpy().andCall(function() {
       return expected;
-    };
-    var result = normalize(predicate, value);
+    });
+    var result = normalize(coercer, value);
     expect(result).toEqual(expected);
-    expect(called).toEqual(true);
+    expect(coercer).toHaveBeenCalled();
+    expect(value).toHaveBeenCalled();
     done();
   });
 
   it('calls the function, passing extra arguments', function(done) {
     var type = 'string';
     var expected = 'test string';
-    var value = function(arg) {
+    var value = expect.createSpy().andCall(function(arg) {
       return arg;
-    };
+    });
     var result = normalize(type, value, expected);
     expect(result).toEqual(expected);
+    expect(value).toHaveBeenCalled();
     done();
   });
 
   it('returns null if result of function does not match type', function(done) {
     var type = 'string';
-    var value = function() {
+    var value = expect.createSpy().andCall(function() {
       return 123;
-    };
+    });
     var result = normalize(type, value);
     expect(result).toEqual(null);
+    expect(value).toHaveBeenCalled();
     done();
   });
 
-  it('returns null if the result of function does not satisfy predicate', function(done) {
-    var predicate = function(value) {
-      return (typeof value === 'string');
-    };
-    var value = function() {
+  it('returns null if return value does not satisfy coercer', function(done) {
```

Comment:
I'm not entirely sure what you mean... The "contract" for custom coercers would be to return a value or `null` to indicate rejection. Are you worried about coercers that do otherwise?